src/Broadway/EventStore/InMemoryEventStore.php

```diff
@@ -53,6 +56,7 @@ public function append($id, DomainEventStreamInterface $eventStream)
             $this->assertPlayhead($this->events[$id], $playhead);
 
             $this->events[$id][$playhead] = $event;
+            $this->allEvents[] = $event;
```

Comment:
> I thought about this more and am realizing it would make it difficult to build a common test case for this if you cannot rely on different implementations returning the events in the same order.

A common test case is possible, but the assertions are a bit more complicated. The assertion would be:
- the _set_ of events returned when visiting is equal to the expected _set_
- the order of the events _per aggregate_ are ordered

I'd be happy to chat on IRC to help you generalize the test cases like this. Shouldn't be too hard.

> I think that in general the idea here is sound, but I'm also thinking this could be problematic if you have a projector that relies on events from multiple aggregate root types. You may run into a case where you won't be able to rebuild the read model in the same way you expect.

Projectors that rely on events from multiple aggregate root types are tricky in any case. If a specific implementation does guarantee to return events in chronological order that's great of course and as a developer you can leverage that fact. It's just hard to guarantee in this library in the general case.

In light of the discussion above I'm conflicted about what to do with the `InMemoryEventStore`. Perhaps it should be configurable? In any case I think we should ship the non-chronological ordered `InMemoryEventStore` in this PR and possibly follow-up with extra things if folks are interested.